Added multimedia images or vidoes to post

alshakerM opened the pull request:
Multi images or videos with next and previous buttons to look like the real IG.
Added play icon to videos.

alshakerM commented:
Renamed StroyVideo component to Video.
Fixed the small posts media dots appearing where there is only one post media.
Fixed the margin of next and previous buttons to match the real IG.

alshakero commented on src/Post/Post.css:
```diff
@@ -67,6 +67,8 @@ section.post-header {
 
 section.img-container {
```
Name can be changed to media-container.

alshakerM replied:
Great catch 👯 
On it

alshakero commented on src/Post/Post.js:
```diff
@@ -359,6 +364,85 @@ function CommentSection({
+  return (
+    <section className="img-container">
+      {media_items[mediaIndex].type === 'photo' ? (
+        <img
+          className="post-img"
+          src={media_items[mediaIndex].url}
+          alt={`${user_name}  Avatar`}
+          loading="lazy"
+        />
```
Didn't we agree to create a component called `PostImage`? Because there will be tags and stuff inside the picture?

alshakero commented on src/Post/Post.js:
```diff
@@ -359,6 +364,85 @@ function CommentSection({
+            onClick={() => setVideoPause(!videoPause)}
+            className="post-video-container"
+          >
+            <Video
+              videoURL={media_items[mediaIndex].url}
+              paused={videoPause}
+              className="post-video"
+              autoPlay={false}
+            />
```
This is good idea but only for a shor time. I'd create another component called `PostVideo` and leave this untouched. Because `StoryVideo` does some time stuff and tracks the progress and so on, and has no Play button. And videos in the post have play buttons. Also the stories have fixed aspect ratio (tall and thin), while post videos come in all sizes.

alshakerM replied:
Awesome suggestion.
Getting it done.

alshakero reviewed (changes requested):
I don't know what happened in this PR, but it's really not as good as your other PRs.

There is no transition when you switch between media items. We mentioned this before when we agreed that you should render all media items next to each other with overflow hidden, just like story avatars, then transform-translate them when the user clicks next.

alshakerM commented:
Switched from index media sliding to map for smoothness and to add transition.

alshakero commented on src/Post/Post.css:
```diff
@@ -65,12 +65,20 @@ section.post-header {
   display: flex;
-  flex-direction: column;
+  width: max-content;
+  overflow-y: hidden;
+  overflow-x: auto;
+  transition: transform 0.25s ease-in-out;
+}
+.media-container::-webkit-scrollbar {
+  display: none;
 }
```
You can hide the scroll by doing `overflow:hidden` instead of `overflow:auto`.
You don't need 

```css
.media-container::-webkit-scrollbar {
  display: none;
}
```

alshakerM replied:
TIL.
Thanks

alshakero commented on src/Post/Post.js:
```diff
@@ -364,76 +364,88 @@ function CommentSection({
+              >
+                <PostVideo
+                  videoURL={mediaItem.url}
+                  paused={videoPause}
```
This means every video in the post will pause and unpause at the same time. I think the pausing state should be inside `PostVideo`. This is a tricky one. We can do it together.

alshakerM replied:
I don't think that's the case here because if you see the real IG when you press the next button or the previous button when the video is playing, the video pauses, which means the buttons need access to the state in order to pause the video.

alshakero replied:
That's correct! But they need access to a specific video, this prop will change for every video.

alshakerM replied:
On it

alshakero commented on src/PostVideo/PostVideo.css:
```diff
@@ -0,0 +1,16 @@
+.story-video {
+  width: 100%;
+  height: 800px;
+  min-width: 449px;
+}
+.play-icon-container {
+  height: 0;
+}
+.play-icon {
+  position: relative;
+  margin-bottom: -50%;
+  left: 50%;
+  color: #fff;
+  transform: scale(3.5);
+  z-index: 3;
+}
```
Why not a CSS module? If you want to keep it like this, you'll have issues since `play-icon` is a common name, and you'll have collisions. With CSS modules, you can use simple names because you're guaranteed to not have collisions.

alshakerM replied:
Done

alshakero reviewed (changes requested):
Much better. Some issues remain. Open an extended post and scroll a bit.

alshakerM commented on src/Post/Post.css:
```diff
@@ -359,3 +369,55 @@ section.comment-input-section {
+  visibility: hidden;
+}
+.next-img-button {
+  margin-left: 529px;
```
TIL.
Done

alshakerM commented on src/Post/Post.css:
```diff
@@ -359,3 +369,55 @@ section.comment-input-section {
+  color: #fff;
+  transform: scale(3.5);
+  z-index: 3;
+}
```
Done

alshakerM commented on src/Post/Post.js:
```diff
@@ -359,6 +364,85 @@ function CommentSection({
+        <img
+          className="post-img"
+          src={media_items[mediaIndex].url}
+          alt={`${user_name}  Avatar`}
```
Great catch.
Thanks

alshakerM commented:
Fixed previous and next media buttons onClick effects.
Switched to module CSS in postVideo and postImg.

alshakero commented:
This issue remains. To reproduce, open an extended post and scroll a bit.
![image](https://user-images.githubusercontent.com/17054134/144604973-5b400b92-2b04-43b6-8705-95120acba4a3.png)

alshakero commented on src/Post/Post.css:
```diff
@@ -359,3 +368,41 @@ section.comment-input-section {
+  background-color: #8e8e8e;
+  border-radius: 50%;
+}
+.progress-dot.is-blue {
+  background-color: rgb(0, 149, 246);
+}
```
When the post is expanded, the progress dot has opacity 0.4, and 0.1 when active. `is-blue` should be is-active or such.

alshakerM commented on src/Post/Post.css:
```diff
@@ -359,3 +368,41 @@ section.comment-input-section {
+  width: 23px;
+  border: none;
+  background: transparent;
+  box-shadow: 2px 2px 6px 1px rgba(0, 0, 0, 0.28);
```
Yea, I didn't understand your comment at first because I didn't know overflow:hidden hides the scroll bar.
TIL

alshakerM replied:
On it

alshakerM commented on src/Post/Post.js:
```diff
@@ -359,6 +364,84 @@ function CommentSection({
     </section>
   );
 }
+function width(el) {
+  return el?.getBoundingClientRect().width;
+}
```
Done

alshakerM commented on src/Post/Post.js:
```diff
@@ -359,6 +364,84 @@ function CommentSection({
+  const mediaContainer = React.useRef();
+  const mediaContainerWidth = width(mediaContainer.current);
+  const [containerWidth, setContainerWidth] = React.useState(0);
+  const [scrollLeft, setScrollLeft] = React.useState(0);
```
Copy that

alshakerM commented:
Fixed the expanded post background when the normal post media take over the screen.
Removed some unneeded CSS.
Added opacity to media buttons.
Moved, a function to utils since it was being used in multiple locations.
Removed some unneeded code.
Changed progress dot is blue name to progress dot is active.
Added progress dot specific design to expanded post.

alshakerM commented:
The title says it all 👯‍♂️

alshakero commented on src/Post/Post.js:
```diff
@@ -359,9 +364,88 @@ function CommentSection({
+        <div className="img-buttons-container">
+          <button
+            className={cx('prev-img-button', { hidden: scrollLeft === 0 })}
+            onClick={() => {
+              setMediaIndex((mediaIndex) => {
+                if (mediaIndex > 0) {
+                  return mediaIndex - 1;
+                }
+              });
+            }}
```
Since the button will be hidden, you don't need to handle the edge cases:
```suggestion
            onClick={() => setMediaIndex(mediaIndex - 1)}
```

alshakero commented on src/Post/Post.js:
```diff
@@ -359,9 +364,88 @@ function CommentSection({
+              hidden: mediaIndex === media_items.length - 1,
+            })}
+            onClick={() => {
+              setMediaIndex((mediaIndex) => {
+                if (mediaIndex < media_items.length - 1) {
+                  return mediaIndex + 1;
+                }
+              });
```
Same a above.

alshakero commented on src/Post/Post.css:
```diff
@@ -359,3 +366,62 @@ section.comment-input-section {
+  justify-content: center;
+  margin-top: 15px;
+}
+.progress-dots-section.is-expanded {
+  display: flex;
+  gap: 5px;
+  justify-content: center;
+  margin: 0;
+  margin-top: -45px;
+  z-index: 2;
+  position: relative;
+}
```
Since `is-expanded` addes more CSS, you don't need to duplicate the code from `.progress-dots-section`
```suggestion
.progress-dots-section.is-expanded {
  margin: 0;
  margin-top: -45px;
  z-index: 2;
  position: relative;
}
```

alshakerM replied:
Done

alshakero reviewed (approved):
So neat now. The app is really looking sharp as hell. Left some cosmetic comments.

alshakerM commented:
Deleted width function from UserStories since I'm using elementWidth from ultis it was pointless.
